cosmos_transfer1/diffusion/inference/transfer_pipeline.py

```diff
@@ -0,0 +1,481 @@
+                if not isinstance(weight, str) or not weight.endswith(".pt"):
+                    try:
+                        # Try converting to float
+                        scalar_value = float(weight)
+                        if scalar_value < 0:
+                            raise ValueError(f"Control weight for {key} must be non-negative.")
+                    except ValueError:
+                        raise ValueError(
+                            f"Control weight for {key} must be a valid non-negative float or a path to a .pt file."
+                        )
+
+    """
+    Naming of the parameters is inline with original CLI and thus the parameters allowed in controlnet_specs.
+    This way we can pass a parameter dictionary directly from the json.
+    """
+
+    def validate_params(
+        self,
+        controlnet_specs,
+        input_video_path=None,
+        prompt=default_prompt,
+        negative_prompt=default_negative_prompt,
+        guidance=5,
+        num_steps=35,
+        seed=1,
+        sigma_max=70.0,
+        blur_strength="medium",
+        canny_threshold="medium",
+        output_dir: str = "outputs/",
+        num_input_frames: int = 1,
+    ):
+        """
+        advanced parameter check
+        """
+        args = argparse.Namespace()
+
+        if sigma_max < 80 and not input_video_path:
+            raise ValueError("Must have 'input_video' specified if sigma_max < 80")
+
+        # Video and prompt settings
+        args_dict = {}
+        if input_video_path:
+            args_dict["input_video"] = input_video_path
+        if prompt:
+            args_dict["prompt"] = prompt
+        if negative_prompt:
+            args_dict["negative_prompt"] = negative_prompt
+
+        # Generation parameters
+        args_dict["guidance"] = guidance
+        args_dict["num_steps"] = num_steps
+        args_dict["seed"] = seed
+        args_dict["sigma_max"] = sigma_max
+        args_dict["blur_strength"] = blur_strength
+        args_dict["canny_threshold"] = canny_threshold
+        args_dict["output_dir"] = output_dir
+        args_dict["num_input_frames"] = num_input_frames
+
+        # validate controlnet_specs
+        self.validate_control_spec(controlnet_specs)
+        args_dict["controlnet_specs"] = controlnet_specs
+        log.info(f"Model parameters: {json.dumps(args_dict, indent=4)}")
+
+        # finally validate interdependencies of the controlnet_specs and input_video_path
+        if not input_video_path:
+            for key in controlnet_specs:
+                if key is "vis" or key is "edge":
+                    raise ValueError(f"Controlnet '{key}' requires an input video. Please specify 'input_video_path'.")
+            else:
+                controlnet = controlnet_specs.get(key)
+                if not controlnet.get("input_control"):
+                    raise ValueError(f"Controlnet '{key}' requires an 'input_control' video OR input_video_path.")
```

Comment:
this block confuses me. if nothing else the `else` on 161 looks incorrectly indented. i'm not sure what the expected behavior is here. maybe the following?

```suggestion
        # finally validate interdependencies of the controlnet_specs and input_video_path
        if not input_video_path:
            for key, controlnet in controlnet_specs.items():
                if key in ["vis", "edge"]:
                    raise ValueError(f"Controlnet '{key}' requires an input video. Please specify 'input_video_path'.")
                elif "input_control" not in controlnet:
                    raise ValueError(f"Controlnet '{key}' requires an 'input_control' video OR input_video_path.")
```